docs/hackathons/ihp_10_2022.rst

```diff
@@ -0,0 +1,285 @@
+
+
+* **handling of numerical parameters will be simplified**. Different algorithms require different numerical parameters. Passing numerical parameters as arguments in methods (e.g. as it is currently done in ``Connection.exp``) is very dirty because it implies API differences or the use of ``**kwargs`` (which becomes unreadable pretty fast). With the new approach we know numerical parameters must be changed in the respective objects (e.g. ``space.metric.exp_solver.n_steps = 20``) and we know what are the numerical parameters we can tweak simply by looking at the documentation of the corresponding object.
+
+
+* **learning algorithms only care about passing the "geometrically" required arguments to "geometric" operations**. Following the previous point, the fact that numerical parameters are controlled via state allows to only pass geometrically meaningful arguments to methods (e.g. ``exp`` will receive ``tangent_vec`` and ``base_point``; ``log`` receives ``point`` and ``base_point``). This means we can set the numerical parameters before passing an equipped space to a learning algorithm. Currently, we have no way of controlling the parameters of ``log``, ``exp``, etc, when we use an equipped space within a learning algorithm.
+
+
+
+* **history will be easier to turn off/on**. For example, in some use cases of parallel transport we are only interested in parallel transporting a tangent vector and we do not want to access the trajectory. In other cases, the trajectory is important. Having a ``ParallelTransporter`` object allows to handle these use cases easily by having a boolean parameter e.g. ``save_trajectory=True`` that turns off/on the saving of the trajectory. If ``False``, all the additional operations required to save the trajectory will not be done (which may lead to better performance and less memory use). If ``True``, the trajectory should be stored within the object (e.g. by creating the attribute ``self.trajectory_``; naming follows ``sklearn`` naming convention), but never be returned, as we want to keep the same API. A similar behavior can be achieved without the creation of an additional object, but then the question of where to save the trajectory arises. "In ``Connection``" may be a OK answer, but it will lead fast to namespace pollution. The same applies to optimization problems, solution of ODEs, etc. This is especially interesting for research.
```

Comment:
The `self.trajectory_` attribute is a bit surprising. I do get the point of not returning the trajectory, but I didn't think of the `ParallelTransporter` as a `sklearn` estimator: the trajectory is specific to the input points and tangent_vecs and cannot be reused with different inputs, while a `coef_` attribute in `sklearn` is reused on new data in `predict`.

Reply:
Maybe the compute method of the `ParallelTransporter` should return an object that has those attributes: `transported_vecs_`, `trajectory_`, etc... and ODE solvers could return an object with the final point and the trajectory if it was required. Kind of what scipy optimizers return and what we are currently doing with a dictionnary actually...